### ISSUE TITLE
cluster: load assignment config from docs for static cluster fails

### ISSUE THREAD

**Issue by yuval-k** [envoyproxy/envoy#4056]
When trying the static example config from here:
https://www.envoyproxy.io/docs/envoy/latest/configuration/overview/v2_overview#static

Envoy fails to load:
```
[2018-08-04 18:27:30.258][24307][critical][main] external/envoy/source/server/server.cc:78] error initializing configuration 'frontcfg.yaml': Proto constraint validation failed (BootstrapValidationError.StaticResources: ["embedded message failed validation"] | caused by StaticResourcesValidationError.Clusters[i]: ["embedded message failed validation"] | caused by ClusterValidationError.LoadAssignment: ["embedded message failed validation"] | caused by ClusterLoadAssignmentValidationError.ClusterName: ["value length must be at least " '\x01' " bytes"]): static_resources {
```

The source of this is the validation here:
https://github.com/envoyproxy/envoy/blob/6b65dbe3a41dea34ab303de52804ddab20556de8/api/envoy/api/v2/eds.proto#L47

And indeed when the issue was solved when the cluster name was added.

They way I understand it, we can either (1)remove the requirement for validation, or (2) fix the docs and add cluster name. Thoughts?

**Comment by dio** [comment]
@yuval-k thanks for catching this. Obviously, we missed testing that. 

Yes, I think adding `cluster_name: name` to the docs will be the fix for now. Since that `cluster_name` is being used in:

https://github.com/envoyproxy/envoy/blob/a614808b9aab7cfd94d66a307a8b65c6c7890097/source/common/upstream/eds.cc#L61-L65

Not directly related, but we can't remove the validation at least for now.

**Comment by yuval-k** [comment]
This bit of code seem to also do validation? If indeed it does, doesn't it make the one in the proto redundant?

**Comment by dio** [comment]
Yes, if the management server sets the corresponding discovery response `cluster_name` to empty/unset it will be failed as well. 

However, we need to communicate this to the management server (developers?) that this `cluster_name` [is required](https://github.com/envoyproxy/go-control-plane/blob/master/envoy/api/v2/eds.pb.validate.go#L45-L50) as part of a discovery response (in this case, for eds).
